x/concentrated-liquidity/keeper_test.go

```diff
@@ -470,32 +475,106 @@ func (s *KeeperTestSuite) runMultiplePositionRanges(ranges [][]int64, rangeTestP
-func (s *KeeperTestSuite) setupPositionsForMinSpotPriceMigration() uint64 {
-	pool := s.PrepareConcentratedPool()
-	poolId := pool.GetId()
+func (s *KeeperTestSuite) setupPositionsForMinSpotPriceMigration(spreadFactor osmomath.Dec) (poolId uint64, positions []cl.CreatePositionData) {
+	pool := s.PrepareCustomConcentratedPool(s.TestAccs[0], ETH, USDC, DefaultTickSpacing, spreadFactor)
+	poolId = pool.GetId()
 
 	// Fund test account with tokens necessary for all positions
 	s.FundAcc(s.TestAccs[0], DefaultCoins.Add(DefaultCoins...).Add(DefaultCoins...))
 
 	// Setup an original full range position
-	s.createDefaultPosition(poolId, types.MinInitializedTick, types.MaxTick)
+	position := s.createDefaultPosition(poolId, types.MinInitializedTick, types.MaxTick)
+	positions = append(positions, position)
 
 	// Setup a full range position on the new min spot price
-	s.createDefaultPosition(poolId, types.MinInitializedTickV2, types.MaxTick)
+	position = s.createDefaultPosition(poolId, types.MinInitializedTickV2, types.MaxTick)
+	positions = append(positions, position)
 
 	// Setup a position between the new min spot price and the old min spot price
-	s.createDefaultPosition(poolId, types.MinInitializedTickV2+1000, types.MinInitializedTick-1000)
+	position = s.createDefaultPosition(poolId, types.MinInitializedTickV2+1000, types.MinInitializedTick-1000)
+	positions = append(positions, position)
+
+	return poolId, positions
+}
+
+// swaps to the lowered minimum tick of 10^-30 and back to the original tick which is in the positive range.
+// There are 3 positions setup over the swapped range (see setupPositionsForMinSpotPriceMigration for details).
+// Additionally, spread factor and incentive rewards are configurable as parameters.
+// Returns pool id position data and actual tokens swapped in (in zero for one direction and back).
+func (s *KeeperTestSuite) swapToMinTickAndBack(spreadFactor osmomath.Dec, incentiveRewards sdk.Coins) (poolId uint64, positions []cl.CreatePositionData, actualAmountsSwappedIn sdk.Coins) {
+	poolId, positions = s.setupPositionsForMinSpotPriceMigration(spreadFactor)
+
+	// Refetch pool
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	incentiveCreator := s.TestAccs[2]
+	s.FundAcc(incentiveCreator, incentiveRewards)
+
+	// Create incentive rewards if desired
+	if !incentiveRewards.Empty() {
+		s.Require().Len(incentiveRewards, 1)
+		incentiveCoin := incentiveRewards[0]
+
+		_, err = s.App.ConcentratedLiquidityKeeper.CreateIncentive(
+			s.Ctx, poolId, s.TestAccs[2],
+			incentiveCoin, incentiveCoin.Amount.ToLegacyDec().Quo(osmomath.NewDec(migrationTestTimeBetweenSwapsSecs)), s.Ctx.BlockTime(), time.Nanosecond)
+		s.Require().NoError(err)
+	}
+
+	// esimate amount in to swap left all the way until the new min initialized tick
+	amountZeroIn, _, _ := s.computeSwapAmounts(poolId, pool.GetCurrentSqrtPrice(), types.MinInitializedTickV2, true, false)
```

Comment:
Do you think it would be worthwhile to also test for swapping to `MinCurrentTickV2`? Or is this not supposed to be possible?

Reply:
Discussed in one of the previous PRs swapping zero for one to min initialized tick bumps us down to min current tick.

Tests were added to prove that this is the case